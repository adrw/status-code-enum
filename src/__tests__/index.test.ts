import deepSortObject from 'deep-sort-object'
import { StatusCode } from '../index'

test('StatusCode', () => {
  expect(StatusCode).toMatchSnapshot()
  expect(JSON.stringify(deepSortObject(StatusCode), undefined, 2)).toEqual(
    JSON.stringify(StatusCode, undefined, 2)
  )
})
