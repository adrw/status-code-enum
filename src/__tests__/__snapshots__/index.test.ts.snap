// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCode 1`] = `
Object {
  "100": "InfoContinue",
  "101": "InfoSwitchingProtocols",
  "102": "InfoProcessing",
  "200": "SuccessOK",
  "201": "SuccessCreated",
  "202": "SuccessAccepted",
  "203": "SuccessNonAuthoritativeInfo",
  "204": "SuccessNoContent",
  "205": "SuccessResetContent",
  "206": "SuccessPartialContent",
  "207": "SuccessMultiStatus",
  "208": "SuccessAlreadyReported",
  "229": "SuccessIMUsed",
  "300": "RedirectMultipleChoices",
  "301": "RedirectMovedPermanently",
  "302": "RedirectFound",
  "303": "RedirectSeeOther",
  "304": "RedirectNotModified",
  "305": "RedirectUseProxy",
  "306": "RedirectSwitchProxy",
  "307": "RedirectTemp",
  "308": "RedirectPermanent",
  "400": "ClientErrorBadRequest",
  "401": "ClientErrorUnauthorized",
  "402": "ClientErrorPaymentRequired",
  "403": "ClientErrorForbidden",
  "404": "ClientErrorNotFound",
  "405": "ClientErrorMethodNotAllowed",
  "406": "ClientErrorNotAcceptable",
  "407": "ClientErrorProxyAuthRequired",
  "408": "ClientErrorRequestTimeout",
  "409": "ClientErrorConflict",
  "410": "ClientErrorGone",
  "411": "ClientErrorLengthRequired",
  "412": "ClientErrorPreconditionFailed",
  "413": "ClientErrorPayloadTooLarge",
  "414": "ClientErrorURITooLong",
  "415": "ClientErrorUnsupportedMediaType",
  "416": "ClientErrorRangeNotSatisfiable",
  "417": "ClientErrorExpectationFailed",
  "418": "ClientErrorImATeapot",
  "421": "ClientErrorMisdirectedRequest",
  "422": "ClientErrorUnprocessableEntity",
  "423": "ClientErrorLocked",
  "424": "ClientErrorFailedDependency",
  "426": "ClientErrorUpgradeRequired",
  "428": "ClientErrorPreconditionRequired",
  "429": "ClientErrorTooManyRequests",
  "431": "ClientErrorRequestHeaderFieldsTooLarge",
  "440": "ClientErrorLoginTimeOut",
  "449": "ClientErrorRetryWith",
  "451": "ClientErrorUnavailableForLegalReasons",
  "500": "ServerErrorInternal",
  "501": "ServerErrorNotImplemented",
  "502": "ServerErrorBadGateway",
  "503": "ServerErrorServiceUnavailable",
  "504": "ServerErrorGatewayTimeout",
  "505": "ServerErrorHTTPVersionNotSupported",
  "506": "ServerErrorVariantAlsoNegotiates",
  "507": "ServerErrorInsufficientStorage",
  "508": "ServerErrorLoopDetected",
  "509": "ServerErrorBandwidthLimitExceeded",
  "510": "ServerErrorNotExtended",
  "511": "ServerErrorNetworkAuthRequired",
  "ClientErrorBadRequest": 400,
  "ClientErrorConflict": 409,
  "ClientErrorExpectationFailed": 417,
  "ClientErrorFailedDependency": 424,
  "ClientErrorForbidden": 403,
  "ClientErrorGone": 410,
  "ClientErrorImATeapot": 418,
  "ClientErrorLengthRequired": 411,
  "ClientErrorLocked": 423,
  "ClientErrorLoginTimeOut": 440,
  "ClientErrorMethodNotAllowed": 405,
  "ClientErrorMisdirectedRequest": 421,
  "ClientErrorNotAcceptable": 406,
  "ClientErrorNotFound": 404,
  "ClientErrorPayloadTooLarge": 413,
  "ClientErrorPaymentRequired": 402,
  "ClientErrorPreconditionFailed": 412,
  "ClientErrorPreconditionRequired": 428,
  "ClientErrorProxyAuthRequired": 407,
  "ClientErrorRangeNotSatisfiable": 416,
  "ClientErrorRequestHeaderFieldsTooLarge": 431,
  "ClientErrorRequestTimeout": 408,
  "ClientErrorRetryWith": 449,
  "ClientErrorTooManyRequests": 429,
  "ClientErrorURITooLong": 414,
  "ClientErrorUnauthorized": 401,
  "ClientErrorUnavailableForLegalReasons": 451,
  "ClientErrorUnprocessableEntity": 422,
  "ClientErrorUnsupportedMediaType": 415,
  "ClientErrorUpgradeRequired": 426,
  "InfoContinue": 100,
  "InfoProcessing": 102,
  "InfoSwitchingProtocols": 101,
  "RedirectFound": 302,
  "RedirectMovedPermanently": 301,
  "RedirectMultipleChoices": 300,
  "RedirectNotModified": 304,
  "RedirectPermanent": 308,
  "RedirectSeeOther": 303,
  "RedirectSwitchProxy": 306,
  "RedirectTemp": 307,
  "RedirectUseProxy": 305,
  "ServerErrorBadGateway": 502,
  "ServerErrorBandwidthLimitExceeded": 509,
  "ServerErrorGatewayTimeout": 504,
  "ServerErrorHTTPVersionNotSupported": 505,
  "ServerErrorInsufficientStorage": 507,
  "ServerErrorInternal": 500,
  "ServerErrorLoopDetected": 508,
  "ServerErrorNetworkAuthRequired": 511,
  "ServerErrorNotExtended": 510,
  "ServerErrorNotImplemented": 501,
  "ServerErrorServiceUnavailable": 503,
  "ServerErrorVariantAlsoNegotiates": 506,
  "SuccessAccepted": 202,
  "SuccessAlreadyReported": 208,
  "SuccessCreated": 201,
  "SuccessIMUsed": 229,
  "SuccessMultiStatus": 207,
  "SuccessNoContent": 204,
  "SuccessNonAuthoritativeInfo": 203,
  "SuccessOK": 200,
  "SuccessPartialContent": 206,
  "SuccessResetContent": 205,
}
`;
